### ISSUE TITLE
Winery engines ordering

### ISSUE THREAD

**Issue by vitor251093** [vitor251093/wineskin#33]
Based in the Discord discussion, I've added the ordering system that is present today in PK to Winery :)

Like we discussed, I've added the change to a branch instead of merging it directly.

**Comment by Gcenx** [comment]
Nice I will add that into my current branches when I'm at my Mac.

Nice, I made a note in the pull-request my commit history is a complete mess and this new branch now it makes sense that I close that fix my commit history then do a new pull-request to the new branch.

It will also include structure changes to the Master Wrapper, since we dropped WineskinX11 I removed all the bash scripts that it needed etc, plus with that removal the needed dylibs shrunk by alot

**Comment by vitor251093** [comment]
> Nice, I made a note in the pull-request my commit history is a complete mess and this new branch now it makes sense that I close that fix my commit history then do a new pull-request to the new branch.

Okidoki ;)

> It will also include structure changes to the Master Wrapper, since we dropped WineskinX11 I removed all the bash scripts that it needed etc, plus with that removal the needed dylibs shrunk by alot

That's great news :D

**Comment by Gcenx** [comment]
I just tested it and it says my engines are all corrupted, also shows file extensions within the name

Not too sure if it's an error with the ordering code or an issue on my end when I merged it into my devel branch

**Comment by vitor251093** [comment]
Sorry, it was a mistake of mine. Just fixed.

**Comment by Gcenx** [comment]
> Sorry, it was a mistake of mine. Just fixed.

That got it, now next question is what Branch do you want me to push my Squashed Development Branch too?

**Comment by Gcenx** [comment]
@vitor251093 another thing that you should add is if XQuartz is not installed only show Mac Driver compatible engines, since we don't want people saying it does not work since they didn't read that they needed to install XQuartz

**Comment by vitor251093** [comment]
> If you added README.md to .gitignore wouldn't that stop the merging of my edited README.md?
Uhn, maybe. I will try that.

> @vitor251093 another thing that you should add is if XQuartz is not installed only show Mac Driver compatible engines, since we don't want people saying it does not work since they didn't read that they needed to install XQuartz
Certainly there must be a difference, but how about this: when someone tried to create a port with engine that is only XQuartz compatible, and XQuartz isn't installed, we show the user a warning instead, with the link to XQuartz's website. 

What do you think?

**Comment by Gcenx** [comment]
> Certainly there must be a difference, but how about this: when someone tries to create a port with an engine that is only XQuartz compatible, and XQuartz isn't installed, we show the user a warning instead, with the link to the XQuartz's website.

That sounds like an option, I'm thinking with the whole don't list them idea since after Mojave we might not even have access to XQuartz anymore. So only showing compatible engines came more to my mind.

**Comment by vitor251093** [comment]
Still too harsh. People may don't have internet available in the first time.

If macOS 10.15/11.0 kills XQuartz, we can make Winery remove those items from the list specifically in macOS 10.15+/11.0+.

**Comment by Gcenx** [comment]
That’s true I just didn’t want it to be annoying. But we still need to stop listing engines that can not be used. Since that will cause headaches.

**Comment by vitor251093** [comment]
Certainly, although we still need to show those engines, to show that they are still option, even if they have special requirements. 

We could make engines that have requirements red, or orange, to differ them from the rest. If people try to install red/orange engines, the warning with their requirement is shown (like that the engine required XQuartz, in those cases). 

However, we can also give the option to hide those engines, with a checkbox for example, and it can be enabled by default.

**Comment by Gcenx** [comment]
I agree 100% with that way of handling it.
Red should be the color since most people should understand Red means stop/error/warning, and having a checkbox being enabled by default and warning message should solve any potential user error.

**Comment by Gcenx** [comment]
Looks very good, I just won't dare to start using that as my daily driver for now.

I will pull this and 7za change into my offline branch to check everything is playing nice.
I already mostly have the #37 changes done for Wineskin.app I'm just doing some basic bug testing. And I might have found another fun bug for me to squash I will post an issue if it turns out to be a Wineskin issue.
